From 524e24b4cbd07db01b9a19268ed8bf2df811b7af Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Tue, 14 Apr 2020 23:29:29 -0700
Subject: [PATCH] Makefile: Pass HOSTLDFLAGS to has_libelf invocation

HOSTCC calls its default linker, which is ld when clang is compiled with
default options. To force using ld.lld, '-fuse-ld=lld' must be passed,
which can be done through our script using HOSTLDFLAGS.

This allows full hermeticity when CONFIG_STACK_VALIDATION is turned on.

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
Signed-off-by: Diab Neiroukh <lazerl0rd@thezest.dev>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 550f753691f4..0069303e5681 100644
--- a/Makefile
+++ b/Makefile
@@ -1160,7 +1160,7 @@ export mod_sign_cmd
 HOST_LIBELF_LIBS = $(shell pkg-config libelf --libs 2>/dev/null || echo -lelf)
 
 has_libelf = $(call try-run,\
-               echo "int main() {}" | $(HOSTCC) -xc -o /dev/null $(HOST_LIBELF_LIBS) -,1,0)
+               echo "int main() {}" | $(HOSTCC) -xc -o /dev/null $(HOSTLDFLAGS) $(HOST_LIBELF_LIBS) -,1,0)
 
 ifdef CONFIG_STACK_VALIDATION
   ifeq ($(has_libelf),1)
